python/cudf/cudf/dataframe/categorical.py

```diff
@@ -486,6 +486,9 @@ def as_datetime_column(self, dtype, **kwargs):
         )
 
     def _get_decategorized_column(self):
+        if self.null_count == len(self):
+            # self._categories is empty; just return codes
+            return self.cat().codes._column
```

Comment:
Is this working around a bug that should be resolved in the gather call below?

Reply:
Not exactly -- the gather call will be something like `gather([], [0, 1, 1, 2])`, which I'd guess should normally be an error.

This is needed because a null categorical has an empty `categories`. There's "nothing" to gather so we shouldn't go down to the gather call"

```python

In [1]: import pandas as pd                                                                                                                                                                                     

In [2]: a = pd.Series([None, None, None], dtype='category')                                                                                                                                                     

In [3]: print(a)                                                                                                                                                                                                
0   NaN
1   NaN
2   NaN
dtype: category
Categories (0, float64): []
```

Reply:
That still feels like a situation that the gather implementation should handle, no? @jrhemstad thoughts?

Reply:
It could also be hitting nvstrings here as well though.